--- glpk/src/glpapi08.c.orig	2008-08-13 10:00:00.000000000 +0200
+++ glpk/src/glpapi08.c	2008-08-16 16:24:08.000000000 +0200
@@ -349,9 +349,15 @@
       switch (ret)
       {  case 0:
             if (tree->mip->mip_stat == GLP_FEAS)
-            {  if (parm->msg_lev >= GLP_MSG_ALL)
-                  xprintf("INTEGER OPTIMAL SOLUTION FOUND\n");
-               tree->mip->mip_stat = GLP_OPT;
+             { if (ios_relative_gap(tree) > 0) {
+                               if (parm->msg_lev >= GLP_MSG_ALL)
+                                       xprintf("INTEGER FEASIBLE SOLUTION FOUND, GAP SMALLER THAN TOLERANCE\n");
+                               tree->mip->mip_stat = GLP_FEAS;
+               } else {
+                       if (parm->msg_lev >= GLP_MSG_ALL)
+                               xprintf("INTEGER OPTIMAL SOLUTION FOUND\n");
+                       tree->mip->mip_stat = GLP_OPT;
+               }
             }
             else
             {  if (parm->msg_lev >= GLP_MSG_ALL)
--- glpk/src/glpios03.c.orig	2008-08-13 10:00:00.000000000 +0200
+++ glpk/src/glpios03.c	2008-08-16 16:24:08.000000000 +0200
@@ -1346,7 +1346,7 @@
          {  if (tree->parm->msg_lev >= GLP_MSG_DBG)
                xprintf("Relative gap tolerance reached; search terminat"
                   "ed\n");
-            ret = GLP_ETMLIM;
+            ret = 0;
             goto done;
          }
 #endif
