--- glpk/examples/Makefile.in.orig	2013-01-28 01:00:00.000000000 -0800
+++ glpk/examples/Makefile.in	2017-11-29 16:48:39.765526637 -0800
@@ -70,7 +70,7 @@
 am_glpsol_OBJECTS = glpsol.$(OBJEXT)
 glpsol_OBJECTS = $(am_glpsol_OBJECTS)
 glpsol_LDADD = $(LDADD)
-glpsol_DEPENDENCIES = ../src/libglpk.la
+glpsol_DEPENDENCIES = ../src/libcoinglpk.la
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
@@ -207,7 +207,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 INCLUDES = -I$(srcdir)/../src
-LDADD = ../src/libglpk.la
+LDADD = ../src/libcoinglpk.la
 glpsol_SOURCES = glpsol.c
 all: all-am
 
@@ -278,6 +278,7 @@
 	  sed -e 'h;s,^.*/,,;s/$(EXEEXT)$$//;$(transform)' \
 	      -e 's/$$/$(EXEEXT)/' `; \
 	test -n "$$list" || exit 0; \
+	{ test -d "$(DESTDIR)$(bindir)" && test -x "$(DESTDIR)$(bindir)"; } || exit 0; \
 	echo " ( cd '$(DESTDIR)$(bindir)' && rm -f" $$files ")"; \
 	cd "$(DESTDIR)$(bindir)" && rm -f $$files
 
--- glpk/src/Makefile.in.orig	2013-01-28 01:00:00.000000000 -0800
+++ glpk/src/Makefile.in	2017-11-29 16:46:02.780518740 -0800
@@ -263,7 +263,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 include_HEADERS = glpk.h
-lib_LTLIBRARIES = libglpk.la
+lib_LTLIBRARIES = libcoinglpk.la
 libglpk_la_LDFLAGS = -version-info 33:0:0 \
 -export-symbols-regex '^(glp_|_glp_lpx_).*'
 
@@ -467,7 +467,7 @@
 	  echo rm -f $${locs}; \
 	  rm -f $${locs}; \
 	}
-libglpk.la: $(libglpk_la_OBJECTS) $(libglpk_la_DEPENDENCIES) $(EXTRA_libglpk_la_DEPENDENCIES) 
+libcoinglpk.la: $(libglpk_la_OBJECTS) $(libglpk_la_DEPENDENCIES) $(EXTRA_libglpk_la_DEPENDENCIES) 
 	$(libglpk_la_LINK) -rpath $(libdir) $(libglpk_la_OBJECTS) $(libglpk_la_LIBADD) $(LIBS)
 
 mostlyclean-compile:
--- glpk/src/glplpx01.c.orig	2017-11-29 16:59:03.171574531 -0800
+++ glpk/src/glplpx01.c	2017-11-28 13:18:03.394933033 -0800
@@ -520,7 +520,7 @@
       ret = glp_simplex(lp, &parm);
       switch (ret)
       {  case 0:           ret = LPX_E_OK;      break;
-         case GLP_EBADB:
+         case GLP_EBADB:   ret = LPX_E_BADB;    break;
          case GLP_ESING:
          case GLP_ECOND:
          case GLP_EBOUND:  ret = LPX_E_FAULT;   break;
